qutip/solver/nm_mcsolve.py

```diff
@@ -0,0 +1,379 @@
+        - completeness_rtol, completeness_atol : float, float, [1e-5, 1e-8]
+          Parameters used in determining whether the given Lindblad operators
+          satisfy a certain completeness relation. If they do not, an
+          additional Lindblad operator is added automatically (with zero rate).
+
+    seeds : int, SeedSequence, list, [optional]
+        Seed for the random number generator. It can be a single seed used to
+        spawn seeds for each trajectory or a list of seeds, one for each
+        trajectory. Seeds are saved in the result and they can be reused with::
+
+            seeds=prev_result.seeds
+
+    target_tol : float, tuple, list, [optional]
+        Target tolerance of the evolution. The evolution will compute
+        trajectories until the error on the expectation values is lower than
+        this tolerance. The maximum number of trajectories employed is
+        given by ``ntraj``. The error is computed using jackknife resampling.
+        ``target_tol`` can be an absolute tolerance or a pair of absolute and
+        relative tolerance, in that order. Lastly, it can be a list of pairs of
+        (atol, rtol) for each e_ops.
+
+    timeout : float, [optional]
+        Maximum time for the evolution in seconds. When reached, no more
+        trajectories will be computed.
+
+    Returns
+    -------
+    results : :class:`qutip.solver.NmmcResult`
+        Object storing all results from the simulation. Compared to a result
+        returned by ``mcsolve``, this result contains the additional field
+        ``trace`` (and ``runs_trace`` if ``store_final_state`` is set). Note
+        that the states on the individual trajectories are not normalized. This
+        field contains the average of their trace, which will converge to one
+        in the limit of sufficiently many trajectories.
+    """
+    H = QobjEvo(H, args=args, tlist=tlist)
+
+    if len(ops_and_rates) == 0:
+        if options is None:
+            options = {}
+        options = {
+            key: options[key]
+            for key in options
+            if key in MESolver.solver_options
+        }
+        return mesolve(
+            H, state, tlist, e_ops=e_ops, args=args, options=options,
+        )
+    else:
+        ops_and_rates = [
+            _parse_op_and_rate(op, rate, tlist=tlist, args=args or {})
+            for op, rate in ops_and_rates
+        ]
+
+    nmmc = NonMarkovianMCSolver(H, ops_and_rates, options=options)
+    result = nmmc.run(state, tlist=tlist, ntraj=ntraj, e_ops=e_ops,
+                      seed=seeds, target_tol=target_tol, timeout=timeout)
+    return result
+
+
+def _parse_op_and_rate(op, rate, **kw):
+    """ Sanity check the op and convert rates to coefficients. """
+    if not isinstance(op, Qobj):
+        raise ValueError("NonMarkovianMCSolver ops must be of type Qobj")
+    if isinstance(rate, numbers.Number):
+        rate = coefficient.const(rate)
```

Comment:
I would propose using
`rate = ConstantCoefficient(rate)`
or adding `ConstantCoefficient` to the `coefficient_builders` dict and not having a special case here.

Reply:
I didn't add it as a special case because I was a bit worried about constants accidentally being converted to `Coefficient`, which is not what one wants when working with `QobjEvo`.

Let me remove `const` entirely for now and just use `ConstantCoefficient` directly as you suggest.

Reply:
Removed.